Hmac.js

```diff
@@ -48,12 +54,42 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} [options.useCache=true] - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    const cacheKey = `sign-${base64url.encode(data)}`;
```

Comment:
Could turn this into a `let`, and move the calculation inside the `useCache` conditional.  No reason to calculate if not using the cache.

I'm guessing the absolute speed here doesn't matter, but could always use a shorter prefix and simpler hex transformation if it ever does matter. ;-)